### PR TITLE
make external links work

### PR DESCRIPTION
**Pull request by greggman**
This is a work in progress so no review needed yet. Though of course feedback welcome. Of course a different category would be good. I just wanted to see it work so stuck it in. I can add the rest if we've decided to go forward with this.

It's not clear to me if linking to external examples a good thing or not. I think Brandon's examples are great but I once you set the example of linking to external examples then everyone will want their sample included. Maybe that's a good thing though?

#362

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -84,6 +84,11 @@ function setSourceTabHash(event: PointerEvent, sourceInfo: SourceInfo) {
   setSourceTab(sourceInfo);
 }
 
+// Non authoritative test that url is for same domain
+function isSameDomain(url: string) {
```
Comment (kainino0x) [inline src/main.ts:88]: nit: I think `new URL(url, window.location).origin === window.location.origin` should work

Comment (kainino0x) [reply src/main.ts:88]: Alternate nit: rename isSameDomain to isRelativeURL or similar

**Comment by greggman** [comment]
okay, updated with all 5 examples. Please review.

File: src/main.ts (hunk 1 above)
Comment (greggman) [reply src/main.ts:93]: choose the first.

**Comment by greggman** [comment]
I agree. I was trying to think of some but I don't know any off the top of my head

I could put a few from webgpufundamentals but I could just copy them here. One that comes to mind is the multiple canvas example. It could use better models on the sample site whereas on the fundamentals site I wanted to keep it simple.

https://webgpufundamentals.org/webgpu/webgpu-multiple-canvases-x200-optimized.html

Another is points but again, can just copy it

https://webgpufundamentals.org/webgpu/webgpu-points-w-size.html

And maybe give it some GUI options for the rest

https://webgpufundamentals.org/webgpu/webgpu-points-w-rotation.html

https://webgpufundamentals.org/webgpu/webgpu-points-3d.html

https://webgpufundamentals.org/webgpu/webgpu-points-3d-fixed-size.html

**Pull request merged**